src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -574,11 +574,38 @@ void ResumptionDataProcessor::AddCommands(
 }
 
 void ResumptionDataProcessor::DeleteCommands(ApplicationSharedPtr application) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const uint32_t app_id = application->app_id();
+  resumption_status_lock_.AcquireForReading();
+  std::vector<ResumptionRequest> failed_requests;
+  if (resumption_status_.find(app_id) != resumption_status_.end()) {
+    failed_requests = resumption_status_[app_id].error_requests;
+  }
+  resumption_status_lock_.Release();
+
+  uint32_t cmd_id_of_failed_VR_command;
```

Comment:
@ychernysheva I believe all this loop may be replaced with `find_if` + lambda

Reply:
@AKalinich-Luxoft Please look at 9a557548570d50dfb3cba68c3cf205bd8da54b3e